JSDemos/Demos/TreeList/LocalReordering/Angular/app/app.component.ts

```diff
@@ -55,7 +55,11 @@ export class AppComponent {
             let sourceIndex = this.employees.indexOf(sourceData),
                 targetIndex = this.employees.indexOf(targetData);
 
-            if (sourceData.Head_ID !== targetData.Head_ID) {
+            if (e.component.isRowExpanded(targetData.ID)) {
+                sourceData.Head_ID = targetData.ID;
+                targetIndex = 0;
+            }
+            else if (sourceData.Head_ID !== targetData.Head_ID) {
```

Comment:
Unfortunately, this does not resolve the issue. 

Steps to reproduce:
1. Expand "COO" -> "Shipping Manager"
2. Drop "Robin Cosworth" (the last Shipping Assistant) between "COO" and "Shipping Manager". This item should be placed between "COO" and "Shipping Manager", but it will be placed right under "Shipping Manager".